lib/factory_bot/definition_proxy.rb

```diff
@@ -176,6 +176,48 @@ def trait(name, &block)
       @definition.define_trait(Trait.new(name, &block))
     end
 
+    # Allows for creation of traits for enumerable values. The instance can be
+    # built with the following allowed syntax.
+    # ----
+    # Default Example:
+    #   factory :task do
+    #     traits_for_enum :status
+    #   end
+    #
+    # Arguments:
+    # * name_of_enum_type_attribute: +Symbol+
+    #   The name of this attribute.
+    # * options: +Hash+
+    # * Where the hash has enum values
+    #   status: { queued: 0, started: 1, finished: 2 }
+    #
+    # ----
+    # Example using a list of values as an Array, or custom enum:
+    #   factory :task do
+    #     traits_for_enum :status, statuses
+    #   end
+    #
+    # Arguments:
```

Comment:
I am finding it a little confusing to have multiple argument lists. Elsewhere I think we have consolidated them into one.

I am thinking:

```
# Arguments:
#   attribute_name: +Symbol+ or +String+ 
#     the name of the attribute these traits will set the value of
#   values: +Array+, +Hash+, or other +Enumerable+
#     An array of trait names, or a mapping of trait names to values for those traits. When this argument is not
#     provided, factory_bot will attempt to get the values from the class of the object being built by calling the
#     pluralized `attribute_name` class method.

Reply:
It makes sense to consolidate, I was questioning whether or not having multiple argument lists would be valuable.